### PR TITLE
add md5 config

### PR DESCRIPTION
**Pull request by srchase**
Adds md5 hashing constructor to runtime configs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptDependency.java [hunk 1]
```diff
@@ -43,6 +43,8 @@
     AWS_CRYPTO_SHA256_BROWSER("dependencies", "@aws-crypto/sha256-browser", "^0.1.0-preview.1", true),
     AWS_SDK_HASH_NODE("dependencies", "@aws-sdk/hash-node", "^0.1.0-preview.5", true),
 
+    AWS_SDK_MD5_BROWSER("dependencies", "@aws-sdk/md-js", "^0.1.0-preview.8", true),
```
Comment (kstich) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptDependency.java]: Should this be `@aws-sdk/md5-js` - notably with the `5`?

The browser config update has this:

```
import { Md5 } from "@aws-sdk/md5-js"
```

Comment (srchase) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptDependency.java]: yes. fixed in e0fd36a65e8134f3039db2357d3284034f7d3241

**Comment by srchase** [comment]
Currently, only S3 and SQS need md5 dependencies.  This should potentially be updated to only pull in md5 dependencies on clients that need them.

**Comment by mtdowling** [comment]
If this is client specific, then it should go in the  AWS SDK for JS v3 repo.

**Pull request closed without merging**